### PR TITLE
code improvements

### PR DESCRIPTION
**Pull request by akoken**
* Unused methods removed
* Used nameof (...) instead of "..."
* Default constructors removed. Compiler already generates it by default.
* Pascal case for constant variables
* readonly where possible

### CODE REVIEW

File: Docker.DotNet/Endpoints/MiscellaneousOperations.cs [hunk 1]
```diff
@@ -21,18 +21,18 @@ public Task AuthenticateAsync(AuthConfigParameters authConfig)
...
             if (authConfig == null)
             {
-                throw new ArgumentNullException("authConfig");
+                throw new ArgumentNullException(nameof(authConfig));
             }
             var data = new JsonRequestContent<AuthConfigParameters>(authConfig, this.Client.JsonSerializer);
 
-            const string path = "auth";
-            return this.Client.MakeRequestAsync(this.Client.NoErrorHandlers, HttpMethod.Post, path, null, data);
+            const string Path = "auth";
```
Comment (jterry75) [inline Docker.DotNet/Endpoints/MiscellaneousOperations.cs]: Why declare the const at all? We could just put it in MakeRequestAsync and have the same "const" without an extra assignment correct?

**Comment by jterry75** [comment]
@akoken - Thanks for the submission! I like the cleanup. Just left one comment that makes sense at all instances of the question.

File: Docker.DotNet/Endpoints/MiscellaneousOperations.cs (hunk 1 above)
Comment (akoken) [inline Docker.DotNet/Endpoints/MiscellaneousOperations.cs]: Absolutely you're right. I looked over the codebase, so didn't noticed. I will make the changes, Thanks @jterry75 !

**Comment by jterry75** [comment]
LGTM. - Mind squashing your changes into 1?

Thanks again!

**Comment by ahmetalpbalkan** [comment]
@jterry75 FYI you can squash yourself while merging and edit the commit message:

image

**Comment by jterry75** [comment]
@ahmetalpbalkan - Fun fact. Thats awesome!

**Pull request merged**